.azure-pipelines.yml

```diff
@@ -0,0 +1,248 @@
+      displayName: Cargo build debug
+      condition: ne( variables['build_type'], 'release' )
+    - bash: RUST_BACKTRACE=1 cargo test --all --exclude lightbeam --exclude wasmtime-wasi-c
+      displayName: Cargo test debug
+      condition: ne( variables['build_type'], 'release' )
+    - script: cargo build --release --all --exclude lightbeam --exclude wasmtime-wasi-c
+      displayName: Cargo build release
+      condition: eq( variables['build_type'], 'release' )
+    - bash: RUST_BACKTRACE=1 cargo test --release --all --exclude lightbeam --exclude wasmtime-wasi-c
+      displayName: Cargo test release
+      condition: eq( variables['build_type'], 'release' )
+
+    # Build and test lightbeam if we're using the nightly toolchain
+    - script: cargo build --package lightbeam
+      displayName: Cargo build lightbeam debug
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), ne( variables['build_type'], 'release' ))
+    - bash: RUST_BACKTRACE=1 cargo test --package lightbeam
+      displayName: Cargo test lightbeam debug
+      # Lightbeam tests fail right now, but we don't want to block on that.
+      continueOnError: true
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), ne( variables['build_type'], 'release' ))
+    - script: cargo build --release --package lightbeam
+      displayName: Cargo build lightbeam release
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), eq( variables['build_type'], 'release' ))
+    - bash: RUST_BACKTRACE=1 cargo test --release --package lightbeam
+      displayName: Cargo test lightbeam release
+      # Lightbeam tests fail right now, but we don't want to block on that.
+      continueOnError: true
+      condition: and(eq(variables['rustup_toolchain'], 'nightly'), eq( variables['build_type'], 'release' ))
+
+    - script: sccache --show-stats
+
+    # Only run on macOS builders, as those are fastest
+    - bash: |
+        cargo doc
+      displayName: Test docs generation
+      condition: and(eq( variables['Agent.OS'], 'Darwin' ), eq( variables['build_type'], 'release' ))
+
+    - bash: |
+        mkdir $BUILD_BINARIESDIRECTORY/bundle
+        cp target/release/{wasmtime,wasm2obj}.exe $BUILD_BINARIESDIRECTORY/bundle
+      displayName: Copy binaries (Win)
+      condition: and(eq( variables['Agent.OS'], 'Windows_NT' ), eq( variables['build_type'], 'release' ))
+    - bash: |
+        mkdir $BUILD_BINARIESDIRECTORY/bundle
+        cp target/release/{wasmtime,wasm2obj} $BUILD_BINARIESDIRECTORY/bundle
+      displayName: Copy binaries (*nix)
+      condition: and(ne( variables['Agent.OS'], 'Windows_NT' ), eq( variables['build_type'], 'release' ))
+    - task: ArchiveFiles@2
+      inputs:
+        rootFolderOrFile: $(Build.BinariesDirectory)/bundle
+        includeRootFolder: false
+        archiveType: 'zip'
+        archiveFile: '$(Build.ArtifactStagingDirectory)/wasmtime-windows-64bit-$(tagName).zip'
```

Comment:
Naming-scheme-wise I personally like how Rust ended up with:

```
$thing-$tag-$target.tar.gz
```

such as

```
wasmtime-master-x86_64-pc-windows-msvc.tar.gz
```

since that allows ARM64, for example, to not conflict with "windows-64bit" which is ambiguous if more support is added.

Reply:
I wanted the names to be as simple as possible, but yeah: the Arm/x64 collision is unfortunate, as we do definitely want to support Arm64 (and 32, and x86-32) soon/